cdap-tms/src/main/java/io/cdap/cdap/messaging/client/DelegatingMessagingService.java

```diff
@@ -139,13 +142,25 @@ private MessagingService getDelegate() {
       LOG.info("Messaging service {} is loaded", messagingService.getName());
       try {
         messagingService.initialize(new DefaultMessagingServiceContext(this.cConf));
-      } catch (IOException e) {
+        createTopics(cConf, messagingService);
```

Comment:
For TMS implementation, it is part of the start up which is called when MessagingService pod comes up : [CoreMessagingService.java#L239](https://github.com/cdapio/cdap/blob/develop/cdap-tms/src/main/java/io/cdap/cdap/messaging/service/CoreMessagingService.java#L239)

I was planning to do createTopics() in SpannerMessagingService.initialise() but the issue here is the MessagingServiceUtils class is also in cdap-tms. Also, DefaultTopicMetadata which is input to the createTopic() method. This class is in cdap-tms module (same level as the DefaultMessagingServiceContext, DelegatingMessagingService). 
We can somehow get away with the DefaultTopicMetadata dependency, as the properties which are set here are not very useful in case of Cloud Spanner. But the MessagingServiceUtils is needed here.

I have added some code comments in the createTopics(), may be, I can add a TODO as well there stating this method can be removed from DelegatingMessagingService after the cdap-tms module refactoring.

Reply:
I have updated code a bit, to just extract TopicMetadata list in Delegating service and then to createTopics() in Spanner implementation.
PTAL.